### ISSUE TITLE
[tools] Add a way to add assemblies to be registered

### ISSUE THREAD

**Issue by rolfbjarne** [xamarin/xamarin-macios#11309]
VSMac needs to run the static registrar for assemblies that aren't referenced by the executable assembly.

Proposal (not very well thought out yet):

1. Add an MSBuild item group:

```xml
<ItemGroup>
    <RegisterStatically Include="path/to/assembly.dll" />
</ItemGroup>
```

2. Pass those assemblies to mtouch/mmp's static registrar during the build.

**Comment by Therzok** [comment]
Yup, an ItemGroup that we can write assemblies to would be great. Maybe something like below?

```xml
<StaticRegistrarReference Include="path/to/assembly.dll" Link="relative/dir" />
```

This would help avoid having a separate target that copies these assemblies to the app bundle (it would make no sense to have the assembly registered and not copied to the resulting app bundle).

Does the Static Registrar care about app.config/binding redirects?

**Comment by rolfbjarne** [comment]
@Therzok can you guys try this package and see if it works as expected: [xamarin.mac-7.11.2.10.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/issue-11309-vsmac/371e812eb765a07c6c040319f71154c36920231c/4862033/package/xamarin.mac-7.11.2.10.pkg)?

The API is as you suggested:

```xml
<StaticRegistrarReference Include="path/to/assembly.dll" Link="relative/dir" />
```

**Comment by Therzok** [comment]
Will test it this week

**Comment by rolfbjarne** [comment]
Closing, VSM doesn't need this anymore.
